client/src/main/java/io/pravega/client/stream/impl/SegmentSelector.java

```diff
@@ -82,9 +83,20 @@ public Segment getSegmentForEvent(String routingKey) {
     public List<PendingEvent> refreshSegmentEventWritersUponSealed(Segment sealedSegment, Consumer<Segment>
             segmentSealedCallback) {
         StreamSegmentsWithPredecessors successors = Futures.getAndHandleExceptions(
-                controller.getSuccessors(sealedSegment), RuntimeException::new);
-        return updateSegmentsUponSealed(currentSegments.withReplacementRange(successors), sealedSegment,
-                segmentSealedCallback);
+                controller.getSuccessors(sealedSegment), t -> {
+                    log.error("Error while fetching successors for segment: {}", sealedSegment, t);
+                    return null;
+                });
+
+        if (successors == null) {
+            // Stream is deleted, complete all pending writes exceptionally.
+            log.error("Pending writes for Segment: {} completed with NoSuchSegmentException", sealedSegment);
+            removeWriter(sealedSegment).forEach(event -> event.getAckFuture()
+                    .completeExceptionally(new NoSuchSegmentException(sealedSegment.toString())));
```

Comment:
If the stream is deleted, then the event writer cannot keep going, how are we exposing this to the application?

Reply:
The writer future will complete with NoSuchSegmentException. Added an integration test case to verify this behavior.

Reply:
Right, but I'm asking about the high-level behavior, what's it that we are exposing to the application? I suppose we are throwing some runtime exception.

Reply:
Apart from completing the write future with an exception the subsequent `writer.writeEvent(..)` will cause a `RuntimeException()` to be thrown to the application.
Note: `writer.flush()` will not throw any exception to the application.